language-support/ts/daml-ledger/index.ts

```diff
@@ -427,6 +427,45 @@ class Ledger {
     return [exerciseResult, events];
   }
 
+  /**
+   * Create a contract for a given template, then immediately exercise a choice
+   * on the created contract.
+   *
+   * @param template The template of the contract to be created.
+   * @param payload The template arguments for the contract to be created.
+   * @param choice The choice to exercise.
+   * @param argument The choice arguments.
+   *
+   * @typeparam T The contract template type.
+   * @typeparam K The contract key type.
+   * @typeparam I The contract id type.
+   * @typeparam C The type of the contract choice.
+   * @typeparam R The return type of the choice.
+   *
+   * @returns The return value of the choice together with a list of
+   * [[event]]'s that includes the creation event for the created contract as
+   * well as all the events that were created as a result of exercising the
+   * choice, including the archive event for the created contract if the choice
+   * is consuming (or otherwise archives it as part of its execution).
+   *
+   */
+  async createAndExercise<T extends object, K, I extends string, C, R>(template: Template<T, K, I>, payload: T, choice: Choice<T, C, R>, argument: C): Promise<[R, Event<object>[]]> {
```

Comment:
Afaict, we can get the `Template` from the choice using `.template()`. That’s how `exercise` is currently implemented. With that we could also follow the same argument order we have for `exercise`, i.e., `Choice<T,C,R>`, contract (here the payload, for exercise the contract id), choice argument. wdyt?

Reply:
I must say I hadn't thought that much about it; I just went through the arguments of create, then the arguments of exercise. Now that you mention it, you're right that the template seems a bit redundant, especially if we see `createAndExercise` as primarily targeted towards the "remote procedure call" use-case.

I like the parallel with `exercise` as this would end up just replacing the contract reference with an explicit contract value. I'll make the change.